driver-core/src/test/java/com/datastax/driver/core/ShardAwareTest.java

```diff
@@ -165,4 +166,36 @@ public void verify_same_shard_in_tracing() throws InterruptedException {
     createData(session);
     this.queryData(session, true);
   }
+
+  @Test(groups = {"short", "scylla"})
+  @CCMConfig(
+      config = {
+        "ring_delay_ms:10000",
+        "native_transport_port:0",
+        "native_shard_aware_transport_port:19042"
+      })
+  public void verify_same_shard_in_tracing_using_only_shard_aware_port()
+      throws InterruptedException {
+
+    LoadBalancingPolicy loadBalancingPolicy = new TokenAwarePolicy(new RoundRobinPolicy());
+
+    ReconnectionPolicy reconnectionPolicy = new ConstantReconnectionPolicy(1 * 1000);
+
+    // We pass only the first host as contact point, so we know the control connection will be on
+    // this host
+    Cluster cluster =
+        register(
+            Cluster.builder()
+                .addContactPoints(getContactPoints().get(0))
+                .withPort(19042)
+                .withReconnectionPolicy(reconnectionPolicy)
+                .withLoadBalancingPolicy(loadBalancingPolicy)
+                .build());
+    cluster.init();
```

Comment:
Why is this needed? It does not seem to be necessary https://github.com/datastax/java-driver/tree/3.x/manual#quick-start